game/src/info/trip.rs

```diff
@@ -884,7 +884,13 @@ fn make_trip_details(
         col.push("Map edits have disconnected the path taken before".text_widget(ctx));
     }
     col.extend(elevation);
-    draw_problems(ctx, app, details, trip_id);
+
+    let analytics = if app.has_prebaked().is_none() || open_trip.show_after {
+        app.primary.sim.get_analytics()
+    } else {
+        app.prebaked()
+    };
+    draw_problems(ctx, app, analytics, details, trip_id);
```

Comment:
I also snuck this in here. It seems reasonable that if we're going to update the route when toggling "before vs. after" we should also update the problems along the route. 

Previously we'd always show the "after" problems, even over top of the "before" route.